distributed/dashboard/scheduler.py

```diff
@@ -319,6 +322,34 @@ def get(self):
         self.set_header("Content-Type", "text/plain")
 
 
+class EventsHandler(WebSocketHandler):
+    def initialize(self, server=None, extra=None):
+        self.server = server
+        self.extra = extra or {}
+        self.plugin = WebsocketPlugin(self, server)
+        self.server.add_plugin(self.plugin)
+
+    def send(self, name, data):
+        data["name"] = name
+        for k in list(data):
+            # Drop bytes objects for now
+            if isinstance(data[k], bytes):
+                del data[k]
+        self.write_message(json.dumps(data))
+
+    def open(self):
+        for worker in self.server.workers:
+            self.plugin.add_worker(self.server, worker)
+
+    def on_message(self, message):
+        message = json.loads(message)
+        if message["name"] == "ping":
+            self.send("pong", {"timestamp": str(datetime.now())})
+
+    def on_close(self):
+        self.server.remove_plugin(self.plugin)
```

Comment:
Do we need to `del self.plugin` to remove the reference cycle, or does Python handle this?

Reply:
I think Python handles this, however I'm not 100% sure.